### DISCUSSION TITLE
C++ collision detection: Use Bullet3Collision, etc. or the Bullet 2.x API?

### DISCUSSION THREAD

**Discussion by wbthomason** [bulletphysics/bullet3#3608] [General]
For a new C++ project using Bullet **only** for collision detection (i.e. no dynamics), is it better to use structures from the `src/Bullet3Collision`, `src/Bullet3Common`, etc. directories, or to use the "old" structures (e.g. `btVector3` instead of `b3Vector3`, `btDbvtBroadphase` instead of `b3DynamicBvhBroadphase`, etc.)?

**Comment by hermestrimegistus** [comment]
I really don't know the difference myself. What is btVector3 and what is b3Vector3? Which is meant for programmer's use and which is there for api designer?

**Comment by phcerdan** [comment]
I would like to clarify this too. @wbthomason did you figure it out?

**Comment by wbthomason** [comment]
@phcerdan unfortunately not; I had to get my project using Bullet finished quickly, so I didn't spend much time after this issue looking into the `Bullet3` options. I just used the structures from the old C++ documentation.

I too would still be interested in clarifying this for using Bullet moving forward.

**Comment by Satya758** [comment]
https://github.com/bulletphysics/bullet3/issues/1401 has information.
